payment_pagseguro/models/payment_transaction.py

```diff
@@ -223,24 +233,62 @@ def _get_pagseguro_charge_params(self):
         # if currency != "BRL":
         #     raise UserError(_("Only BRL currency is allowed."))
         CHARGE_PARAMS = {
-            "reference_id": str(self.payment_token_id.acquirer_id.id),
-            "description": self.display_name[:13],
-            "amount": {
-                # Charge is in BRL cents -> Multiply by 100
-                "value": int(self.amount * 100),
-                "currency": "BRL",
+            "reference_id": self.sale_order_ids[0].name,
+            "customer": {
+                "name": self.partner_name,
+                "email": self.partner_email,
+                "tax_id": punctuation_rm(self.partner_id.vat),
+                "phones": [
+                    {
+                        "country": "55",
+                        "area": "11",
+                        "number": "999999999",
+                        "type": "MOBILE",
+                    }
+                ],
             },
-            "payment_method": {
-                "soft_descriptor": self.acquirer_id.company_id.name,
-                "type": "CREDIT_CARD",
-                "installments": 1,
-                "capture": False,
-                "card": {
-                    "encrypted": self.payment_token_id.pagseguro_card_token,
-                },
+            "items": [
```

Comment:
eu colocaria algo assim para listar as sale_order_line:
items = []
for line in self.sale_order_ids[0].order_line:
    item = {
        "reference_id": line.product_id.default_code,
        "name": line.product_id.name,
        "quantity": line.product_uom_qty,
        "unit_amount": int(line.price_unit * 100),
    }
    items.append(item)

Reply:
@augustodinizl se possível, pega esta branch e submete uma PR para o fork da https://github.com/escodoo/l10n-brazil fazendo favor.